aten/src/ATen/core/jit_type.h

```diff
@@ -1384,9 +1385,42 @@ struct ClassType;
 using ClassTypePtr = std::shared_ptr<ClassType>;
 using ::torch::jit::script::CompilationUnit;
 using ::torch::jit::Function;
+struct SerializableType;
+using SerializableTypePtr = std::shared_ptr<SerializableType>;
+
+struct CAFFE2_API SerializableType : public Type {
+  SerializableType(TypeKind tk, c10::QualifiedName qualifiedName)
+    : Type(tk)
+    , name_(qualifiedName) {}
+
+  std::string python_str() const {
+    return name_.qualifiedName();
+  }
+
+  std::string qualname() const {
+    return name_.qualifiedName();
+  }
+
+  std::string qualifier() const {
+    return name_.prefix();
+  }
+
+  std::string basename() const {
+    return name_.name();
+  }
+
+  virtual std::vector<Function*> methods() const = 0;
```

Comment:
What is the intention here? I don't image NamedTuple or Interfaces will have any methods.

Reply:
Switched to using a `switch`